src/main/java/net/sourceforge/plantuml/project/command/CommandPrintScale.java

```diff
@@ -93,10 +96,16 @@ protected CommandExecutionResult executeArg(GanttDiagram diagram, LineLocation l
 		final String option = arg.get("OPTION", 0);
 		if (option != null)
 			if (option.contains("date"))
-				diagram.setWeeklyHeaderStrategy(WeeklyHeaderStrategy.DAY_OF_MONTH);
-			else if (option.contains("1"))
-				diagram.setWeeklyHeaderStrategy(WeeklyHeaderStrategy.FROM_1);
-
+				diagram.setWeeklyHeaderStrategy(WeeklyHeaderStrategy.DAY_OF_MONTH, 0);
+			else {
+				//int weekStartingNumber = -11;
+				final Pattern p = Pattern.compile("[^-\\d]*(-?\\d+)");
+				final Matcher m = p.matcher(option);
+				if (m.find()) {
+					final int weekStartingNumber = Integer.parseInt(m.group(1));
+					diagram.setWeeklyHeaderStrategy(WeeklyHeaderStrategy.FROM_N, weekStartingNumber);
+				}
+			}
```

Comment:
Hi @arnaudroques 

- How can we directly retrieve the signed integer instead of doing everything I had to add?

I am not so familiar with your internal [plantuml regex classes](https://github.com/plantuml/plantuml/tree/master/src/main/java/net/sourceforge/plantuml/regex).
_If you put me in the way or give any direction..._

Regards,
Th.